fabtests/ubertest/cq.c

```diff
@@ -71,11 +71,14 @@ bool ft_generates_rx_comp(void)
 	if (test_info.test_class & FI_ATOMIC)
 		return false;
 
-	if ((test_info.test_class & FI_RMA) &&
-	    !is_data_func(test_info.class_function) &&
-	    !(is_msg_func(test_info.class_function) &&
-	      test_info.msg_flags & FI_REMOTE_CQ_DATA))
-		return false;
+	if ((test_info.test_class & FI_RMA)) {
+		if (!is_data_func(test_info.class_function) &&
+		    !(is_msg_func(test_info.class_function) &&
+		      test_info.msg_flags & FI_REMOTE_CQ_DATA))
+			return false;
+		if (!(test_info.mode & FI_RX_CQ_DATA))
+			return true;
```

Comment:
This returns true for any RMA operation. Should be put inside the body of the previous `if` statement.

Reply:
We only get here if it is a an RMA operation with CQ data. The check above always returns false for non CQ data operations. If it is an RMA with CQ data and we are not using the RX CQ DATA mode, then we should always write a completion

Reply:
my bad, it is fine.